### PR TITLE
fix: [CDS-73636]: fix NPE for ambiance in pipeline chaining

### PR DESCRIPTION
**Pull request by shalini-agr**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/50148)

### CODE REVIEW

**Comment by shalini-agr** [comment]
trigger uts

File: pipeline-service/service/src/main/java/io/harness/pms/pipelinestage/helper/PipelineStageHelper.java [hunk 1]
```diff
@@ -287,7 +287,9 @@ public PipelineStageOutcome resolveOutputVariables(Map<String, ParameterField<St
 
       resolvedMap.put(entry.getKey(), expression);
     }
-
+    if (ambiance == null) {
```
Comment (mohitg0795) [inline pipeline-service/service/src/main/java/io/harness/pms/pipelinestage/helper/PipelineStageHelper.java:290]: shouldn't this be the first check in this method?

**Comment by shalini-agr** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/pipelinestage/helper/PipelineStageHelper.java` line 290 at r1](https://reviewable.io/reviews/harness/harness-core/50148#gh-1252789714:-NZ_Iw67Ap81OAXTQEGn:b-cahx6k) ([raw file](https://github.com/harness/harness-core/blob/3ab8280604790a8b9ffa1ec9dca8a292ecad9b27/pipeline-service/service/src/main/java/io/harness/pms/pipelinestage/helper/PipelineStageHelper.java#L290)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

shouldn't this be the first check in this method?
</blockquote></details>

map in the input argument is of type Map<String, ParameterField<String>>, but in PipelineStageOutcome, we need  Map<String, Object>, that's why we need to build resolvedMap first.

**Comment by shalini-agr** [comment]
trigger review

**Comment by mohitg0795** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/pipelinestage/helper/PipelineStageHelper.java` line 290 at r1](https://reviewable.io/reviews/harness/harness-core/50148#gh-1252789714:-NZ_ZQUb5HrWBG_MWB6l:b-dwtcly) ([raw file](https://github.com/harness/harness-core/blob/3ab8280604790a8b9ffa1ec9dca8a292ecad9b27/pipeline-service/service/src/main/java/io/harness/pms/pipelinestage/helper/PipelineStageHelper.java#L290)):*
<details><summary><i>Previously, shalini-agr (Shalini Agrawal) wrote…</i></summary><blockquote>

map in the input argument is of type Map<String, ParameterField<String>>, but in PipelineStageOutcome, we need  Map<String, Object>, that's why we need to build resolvedMap first. 
</blockquote></details>

I guess there are some changes now and the comment itself is not valid anymore.

**Review by mohitg0795 — APPROVED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50148)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Comment by shalini-agr** [comment]
trigger uts

**Review by archit-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50148)* status: :shipit: complete! ✓

**Comment by shalini-agr** [comment]
trigger review

**Pull request merged**